### PR TITLE
Add action eventmaterialitemcrafts

### PR DESCRIPTION
**Pull request by tyrosine1153**
### Description

- Add action eventmaterialitemcrafts
- Add ItemMaterialSelectPopup
- Edit Craft for material craft

### Related Links

[아레나 그랜드 피날레 레시피 제작 9c 개발](https://app.asana.com/0/958521740385861/1203404367720310/f)

### Screenshot

image
image
image
image

### CODE REVIEW

File: nekoyume/Assets/_Scripts/State/RxProps.Event.cs [hunk 1]
```diff
@@ -88,10 +96,7 @@ private static void StartEvent()
                 .AddTo(_disposables);
             _eventScheduleRowForRecipe
                 .ObserveOnMainThread()
-                .Subscribe(_ =>
-                {
-                    UpdateEventRecipeRemainingTimeText(_agent.BlockIndex);
-                })
+                .Subscribe(_ => { UpdateEventRecipeRemainingTimeText(_agent.BlockIndex); })
```
Comment (boscohyun) [inline nekoyume/Assets/_Scripts/State/RxProps.Event.cs:99]: 😊
```suggestion
                .Subscribe(_ => UpdateEventRecipeRemainingTimeText(_agent.BlockIndex))
```

File: nekoyume/Assets/_Scripts/UI/Combination/SubRecipeView.cs [hunk 2]
```diff
@@ -140,30 +144,50 @@ private void Awake()
                 });
             }
 
-            button.OnClickSubject
-                .Subscribe(state => { CombineCurrentRecipe(); })
-                .AddTo(gameObject);
+            if (button != null)
+            {
+                button.OnClickSubject
+                    .Subscribe(state => { CombineCurrentRecipe(); })
```
Comment (boscohyun) [inline nekoyume/Assets/_Scripts/UI/Combination/SubRecipeView.cs:150]: 😊
```suggestion
                    .Subscribe(state => CombineCurrentRecipe())
```

File: nekoyume/Assets/_Scripts/UI/Widget/Popup/ItemMaterialSelectPopup.cs [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+                    var itemCount = inventoryItems.FirstOrDefault(item => Equals(item.item.Id, itemBase.Id))?.count ?? 0;
+                    var countableItem = new CountableItem(itemBase, itemCount);
+                    var lastSelectedCount = lastSelectedMaterials?.GetValueOrDefault(id) ?? 0;
+                    return new ItemMaterialSelectScroll.Model(countableItem, Mathf.Min(lastSelectedCount, itemCount));
+                })
+                .OrderByDescending(model => model.Item.Count.Value)
+                .ThenBy(model => model.Item.ItemBase.Value.Id)
+                .ToList();
+            Models = models;
+            ObservableExtensions.Subscribe(Observable.NextFrame(), _ =>
```
Comment (boscohyun) [inline nekoyume/Assets/_Scripts/UI/Widget/Popup/ItemMaterialSelectPopup.cs:81]: If you using `using UniRx;` inside of namespace `Nekoyume.UI`, This line can be using as extension method.

**Review by boscohyun — APPROVED** [review]
🙌

File: nekoyume/Assets/_Scripts/UI/Combination/SubRecipeView.cs [hunk 4]
```diff
@@ -236,6 +273,14 @@ public void SetData(SheetRow<int> recipeRow, List<int> subRecipeIds)
                     recipeCell.Show(consumableRow, false);
                     break;
                 }
+                case EventMaterialItemRecipeSheet.Row materialRow :
+                {
+                    var resultItem = materialRow.GetResultMaterialItemRow();
+                    title = resultItem.GetLocalizedName(false, false);
+                    mainStatTexts.FirstOrDefault()!.text = resultItem.GetLocalizedDescription();
```
Comment (sonohoshi) [inline nekoyume/Assets/_Scripts/UI/Combination/SubRecipeView.cs:280]:
```suggestion
                    mainStatTexts.First().text = resultItem.GetLocalizedDescription();
```

File: nekoyume/Assets/_Scripts/UI/Combination/SubRecipeView.cs [hunk 5]
```diff
@@ -584,6 +640,13 @@ private void UpdateButtonForConsumable()
             lockedObject.SetActive(false);
         }
 
+        private void UpdateButtonForEventMaterial()
+        {
+            materialSelectButton.SetCost(new ConditionalCostButton.CostParam(CostType.NCG, 0));
+            materialSelectButton.Interactable = true;
+            materialSelectButton .gameObject.SetActive(true);
```
Comment (sonohoshi) [inline nekoyume/Assets/_Scripts/UI/Combination/SubRecipeView.cs:647]:
```suggestion
            materialSelectButton.gameObject.SetActive(true);
```

File: nekoyume/Assets/_Scripts/BlockChain/ActionRenderHandler.cs [hunk 6]
```diff
@@ -793,6 +803,36 @@ private void ResponseCombinationConsumable(ActionBase.ActionEvaluation<Combinati
...
 
+        private void ResponseEventMaterialItemCrafts(
+            ActionBase.ActionEvaluation<EventMaterialItemCrafts> eval)
+        {
+            if (eval.Exception is not null)
+            {
+                return;
+            }
+
+            // Todo : ResponseEventMaterialItemCrafts
```
Comment (sonohoshi) [inline nekoyume/Assets/_Scripts/BlockChain/ActionRenderHandler.cs:814]:
```suggestion
```

**Review by sonohoshi — APPROVED** [review]

**Pull request merged**